From 3ef7a7f4227970a4fb048b08d3f1d7bbbd3836ed Mon Sep 17 00:00:00 2001
From: Alan Tull <alan.tull@freescale.com>
Date: Fri, 8 Oct 2010 11:59:38 -0500
Subject: [PATCH] ENGR00132411 sgtl5000 vdda is on regulator sw4

Change the regulator setup to show the correct regulator for vdda.

Signed-off-by: Alan Tull <alan.tull@freescale.com>
---
 arch/arm/mach-mx5/mx50_rdp_pmic_mc13892.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx5/mx50_rdp_pmic_mc13892.c b/arch/arm/mach-mx5/mx50_rdp_pmic_mc13892.c
index e8411ce..b72e987 100644
--- a/arch/arm/mach-mx5/mx50_rdp_pmic_mc13892.c
+++ b/arch/arm/mach-mx5/mx50_rdp_pmic_mc13892.c
@@ -94,12 +94,15 @@ static struct regulator_consumer_supply sw1_consumers[] = {
 	}
 };
 
-static struct regulator_consumer_supply vgen1_consumers[] = {
+static struct regulator_consumer_supply sw4_consumers[] = {
 	{
 		/* sgtl5000 */
 		.supply = "VDDA",
 		.dev_name = "1-000a",
 	},
+};
+
+static struct regulator_consumer_supply vgen1_consumers[] = {
 	{
 		/* sgtl5000 */
 		.supply = "VDDIO",
@@ -171,7 +174,9 @@ static struct regulator_init_data sw4_init = {
 		.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE,
 		.always_on = 1,
 		.boot_on = 1,
-	}
+	},
+	.num_consumer_supplies = ARRAY_SIZE(sw4_consumers),
+	.consumer_supplies = sw4_consumers,
 };
 
 static struct regulator_init_data viohi_init = {
-- 
1.5.4.4

